### ISSUE TITLE
hpx project name is not HPX

### ISSUE THREAD

**Issue by biddisco** [STEllAR-GROUP/hpx#1675]
I have found that scripts for finding HPX in certain projects do not behave as expected.

The problem is caused by the fact that hpx declares itself as
```
    project(hpx C CXX)
```
and generates the file
```
    /lib/cmake/hpx/HPXConfig.cmake
```
which is correctly found by 
```
    find_package(HPX)
```
but not by
```
    find_package(hpx)
```
which is created by scripts which assume the project name is hpx (as declared in the project cmakelists)
The find_package documentation states : 
```
The command searches for a file called "<name>Config.cmake" or \
    "<lower-case-name>-config.cmake" for each name specified.
```

so we should either rename the project to HPX instead of hpx, or modify the gnerated config file to be hpxConfig.cmake. (but that means anyone using find_package(HPX) will have to change their cmake), so option 1 looks better.

**Comment by sithhell** [comment]
Option 1 sounds sane, yes. Can you make that change?

**Comment by hkaiser** [comment]
I agree, let's go for option 1.

**Comment by hkaiser** [comment]
This was fixed by merging #1682
